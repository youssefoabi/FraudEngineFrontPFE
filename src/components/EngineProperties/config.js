// eslint-disable-next-line import/prefer-default-export
export const config = [
  {
    columns: 4,
    id: 'engineName',
    type: 'input',
    label: "Libellé de l'onglet",
  },
  {
    columns: 4,
    id: 'pictogramUrl',
    type: 'input',
    label: 'Url du picto',
  },
  {
    columns: 2,
    id: 'pictogramUrl',
    type: 'imageUrlWithPreview',
    label: 'Aperçu picto',
    imageSize: '40x35',
  },
  {
    columns: 2,
    id: 'isEnable',
    type: 'switch',
    label: 'Onglet actif',
  },
];
