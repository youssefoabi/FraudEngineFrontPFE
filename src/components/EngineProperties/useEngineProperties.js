import { pathOr } from 'ramda';

const useEngineProperties = ({ enginesGroup, handleChange }) => {
  const { engines, currentEngine } = enginesGroup || {};

  const engineId = pathOr(null, ['engineId'], currentEngine);
  const engineName = pathOr('', [engineId, 'engineName'], engines);

  const handleEngineChange = ({ id, value }) => {
    handleChange({
      id: 'engines',
      value: { ...(engines || {}), [engineId]: { ...(engines[engineId] || {}), [id]: value } },
    });
  };

  return {
    engineId,
    engineName,
    handleEngineChange,
  };
};

export default useEngineProperties;
