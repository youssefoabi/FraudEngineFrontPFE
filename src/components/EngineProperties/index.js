import { DataEdit, GridItem } from '@cdiscount/backoffice-ui';
import { func, shape } from 'prop-types';
import { isNilOrEmpty } from 'ramda-adjunct';
import { pathOr } from 'ramda';
import React from 'react';

import { config } from './config';
import useEngineProperties from './useEngineProperties';

const EngineProperties = ({ enginesGroup, handleChange }) => {
  const { engineId, engineName, handleEngineChange } = useEngineProperties({
    enginesGroup,
    handleChange,
  });
  const engine = pathOr(null, ['engines', engineId], enginesGroup);
  const currentEngine = pathOr(null, ['currentEngine'], enginesGroup);

  if (isNilOrEmpty(currentEngine)) return null;

  return (
    <GridItem title={`Propriétés de l'onglet du moteur ${engineName || ''}`}>
      <DataEdit fields={config} values={engine} onChange={handleEngineChange} />
    </GridItem>
  );
};

EngineProperties.propTypes = {
  enginesGroup: shape({}),
  handleChange: func,
};

EngineProperties.defaultProps = {
  enginesGroup: null,
  handleChange: Function.Prototype,
};

export default EngineProperties;
