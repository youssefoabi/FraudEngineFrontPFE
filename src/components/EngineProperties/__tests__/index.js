import { itRendersAllMutations } from '@cdiscount/jest-utils';

import EngineProperties from '../';

jest.resetModules();

jest.mock('../useEngineProperties', () => ({
  __esModule: true,
  default: jest
    .fn()
    .mockReturnValueOnce({})
    .mockReturnValueOnce({})
    .mockReturnValueOnce({
      engineName: 'engine name',
    }),
  namedExport: 'useEngineProperties',
}));

const mutations = [
  { name: 'without props' },
  { name: 'with props and without engineName', props: { enginesGroup: { currentEngine: { engineId: 1 } } } },
  { name: 'with props and engineName', props: { enginesGroup: { currentEngine: { engineId: 1 } } } },
];

describe(EngineProperties.name, () => {
  itRendersAllMutations(EngineProperties, mutations);
});
