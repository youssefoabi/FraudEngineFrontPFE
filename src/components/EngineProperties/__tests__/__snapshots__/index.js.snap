// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EngineProperties EngineProperties renders correctly with props and engineName 1`] = `
<h
  fullheight={false}
  title="Propriétés de l'onglet du moteur engine name"
>
  <f
    config={null}
    fields={
      Array [
        Object {
          "columns": 4,
          "id": "engineName",
          "label": "Libellé de l'onglet",
          "type": "input",
        },
        Object {
          "columns": 4,
          "id": "pictogramUrl",
          "label": "Url du picto",
          "type": "input",
        },
        Object {
          "columns": 2,
          "id": "pictogramUrl",
          "imageSize": "40x35",
          "label": "Aperçu picto",
          "type": "imageUrlWithPreview",
        },
        Object {
          "columns": 2,
          "id": "isEnable",
          "label": "Onglet actif",
          "type": "switch",
        },
      ]
    }
    inputs={null}
    onChange={[Function]}
    onInit={[Function]}
    title={null}
    values={null}
  />
</h>
`;

exports[`EngineProperties EngineProperties renders correctly with props and without engineName 1`] = `
<h
  fullheight={false}
  title="Propriétés de l'onglet du moteur "
>
  <f
    config={null}
    fields={
      Array [
        Object {
          "columns": 4,
          "id": "engineName",
          "label": "Libellé de l'onglet",
          "type": "input",
        },
        Object {
          "columns": 4,
          "id": "pictogramUrl",
          "label": "Url du picto",
          "type": "input",
        },
        Object {
          "columns": 2,
          "id": "pictogramUrl",
          "imageSize": "40x35",
          "label": "Aperçu picto",
          "type": "imageUrlWithPreview",
        },
        Object {
          "columns": 2,
          "id": "isEnable",
          "label": "Onglet actif",
          "type": "switch",
        },
      ]
    }
    inputs={null}
    onChange={[Function]}
    onInit={[Function]}
    title={null}
    values={null}
  />
</h>
`;

exports[`EngineProperties EngineProperties renders correctly without props 1`] = `""`;
